### PR TITLE
Init documentation for the network operators.

### PR DESCRIPTION
**Pull request by alacuku**
A brief description of the network operators.

### CODE REVIEW

File: docs/design/liqonet_tunnelEndpoint.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# TunnelEndpoint-Operator
+## Overview
+TunnelEndpoint is the component in charge to bring up a VPN tunnel with the peering clusters.
```
Comment (fraborg) [inline docs/design/liqonet_tunnelEndpoint.md:3]:
```suggestion
TunnelEndpoint is the component in charge of building up a VPN tunnel with the peering clusters.
```

File: docs/design/liqonet_routeOperator.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Route-Operator
+## Overview
+The Route-Operator runs as a DaemonSet on all Kubernetes nodes, and ensures route rules to allow all local pods/nodes to communicate through the elected gateway node  with the pods of the peering cluster.
```
Comment (frisso) [inline docs/design/liqonet_routeOperator.md:3]: How is the gateway node elected?
Can the operator influence this election?

File: docs/design/liqonet_routeOperator.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Route-Operator
+## Overview
+The Route-Operator runs as a DaemonSet on all Kubernetes nodes and coordinates the setup of all the routes that allow each local pod/node to communicate with the pods of the peering cluster, through the elected gateway node.
```
Comment (frisso) [inline docs/design/liqonet_routeOperator.md:3]: How is the gateway node elected?
Can the operator influence this election?

File: docs/design/liqonet_routeOperator.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+It will ensure state and react on tunnelEndpoint CR changes, which means that it is able to add/remove routes as soon as a new cluster peers/de-peers with the local cluster.
+It creates a VxLan overlay network to whom all the cluster nodes belong and uses it to direct all the network traffic to the gateway node.
+
+### Features
+* Traffic toward remote networks routed through a Vxlan overlay (set-up dynamically)
+* Suppor for Single and Double NATting
```
Comment (frisso) [inline docs/design/liqonet_routeOperator.md:9]: Is the natting under responsibility of this operator? If so, it has to be mentioned in the "overview"

File: docs/design/liqonet_routeOperator.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+* Support for Node-to-(Remote)Pod and Pod-to-(Remote)Pod communication patterns
+* Tested with Flannel but should work with any other CNI plugin (Calico, Cannal, etc.).
+
+### Limitations
+* No support for dynamic gateway node (yet)
+* New nodes added to the cluster are not dynamically added to the VxLan network
```
Comment (frisso) [inline docs/design/liqonet_routeOperator.md:15]: Do you mean "added to the *local* cluster"?

File: docs/design/liqonet_tunEndCreator.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# TunnelEndpointCreator-Operator
+## Overview
+The TunnelEndpointCreator is part of the network module and runs as a deployment. It reacts to **Advertisements Custom Resources (CR)**
```
Comment (frisso) [inline docs/design/liqonet_tunEndCreator.md:3]: If you write in bold CR here, you should write in bold all CRs, also in the other .md files.

File: docs/design/liqonet_tunEndCreator.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+and for each one it creates a **TunnelEndpoint CR**. The advertisement CR carries all the required data
+in order to establish a point-to-point connection with the peering cluster. A simple IP manager
+has been embedded inside the operator in order to resolve the possible conflicts rising between the network
+address spaces used by the clusters. The peering clusters could have address spaces conflicts:
+* pod network CIDR
+* clusterIP network CIDR
+
+### Features
+If the pod network CIDR of a peering cluster does not have conflicts with the address spaces
```
Comment (frisso) [inline docs/design/liqonet_tunEndCreator.md:12]: What you write here is not a list of features. Move in the overview and add a list of features here.

File: docs/design/liqonet_tunEndCreator.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+used by the local network then the IPAM just marks this new network subnet as reserved.
+In the other case the remote pod network CIDR is remapped in a new virtual address space.
+In such way the remapping process happens if necessary. 
+
+
+### Limitations
+Currently the new virtual address spaces belong to the 10.0.0.0/8 CIDR Block hence:
```
Comment (frisso) [inline docs/design/liqonet_tunEndCreator.md:19]: I do not understand the limitations. Could you please be more verbose?

File: docs/design/liqonet_tunEndCreator.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+* each new subnet is a 10.x.x.x/16
+* max 256 new subnets available to remap the peering clusters
```
Comment (frisso) [inline docs/design/liqonet_tunEndCreator.md:21]: What does this mean for an end user? Max 256 peering clusters?

File: docs/design/liqonet_tunnelEndpoint.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+All the traffic between the local and remote cluster is first delivered to the Gateway Node, then tunneled towards the remote gateway and finally delivered to the desired destination.
+The traffic can leave local cluster as it is in case the home and remote addressing spaces do not overlap; otherwise, the traffic crosses a properly configurated NAT in order to avoid overlapped spaces.
+
+The TunnelEndpoint-Operator runs as deployment only on the local Gateway Node.
+
+### Features
+* GRE tunnel as VPN tunnel
+
+### Limitations
+* Does not support other VPN solutions
```
Comment (frisso) [inline docs/design/liqonet_tunnelEndpoint.md:13]: So, it means that supports only GRE tunnels?

File: docs/design/liqonet_tunnelEndpoint.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+* Traffic flows as it is, does not add security
```
Comment (frisso) [inline docs/design/liqonet_tunnelEndpoint.md:14]: Should we write "Unsupported security policies" instead?

File: docs/design/liqonet_tunnelEndpoint.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
+# TunnelEndpoint-Operator
+## Overview
+This component is in charge of establishing a VPN tunnel with the peering clusters, which is created between the (local) Gateway Node and the Gateway Node of the remote cluster.
+All the traffic between the local and remote cluster is first delivered to the Gateway Node, then tunneled towards the remote gateway and finally delivered to the desired destination.
+The traffic can leave the local cluster as is in case the home and remote addressing spaces do not overlap; otherwise, the traffic crosses a properly configurated NAT in order to avoid overlapped spaces.
```
Comment (frisso) [inline docs/design/liqonet_tunnelEndpoint.md:5]: Who configures the NAT?

File: docs/design/liqonet_routeOperator.md (hunk 3 above)
Comment (alacuku) [reply docs/design/liqonet_routeOperator.md:3]: the node is selected at install time labelling the node where the tunneloperator will run.

File: docs/design/liqonet_routeOperator.md (hunk 4 above)
Comment (alacuku) [reply docs/design/liqonet_routeOperator.md:9]: tunnelEndpointCreator is the operator in charge of natting operations.

File: docs/design/liqonet_routeOperator.md (hunk 5 above)
Comment (alacuku) [reply docs/design/liqonet_routeOperator.md:15]: yes, i do. going to add it.

File: docs/design/liqonet_tunEndCreator.md (hunk 9 above)
Comment (alacuku) [reply docs/design/liqonet_tunEndCreator.md:21]: Max 256 peering clusters, but could be less if some of this subnets are reserved/used in the local cluster.

File: docs/design/liqonet_tunnelEndpoint.md (hunk 10 above)
Comment (alacuku) [reply docs/design/liqonet_tunnelEndpoint.md:13]: yes, only GRE tunnels.

File: docs/design/liqonet_tunnelEndpoint.md (hunk 12 above)
Comment (alacuku) [reply docs/design/liqonet_tunnelEndpoint.md:5]: The iptables rules configuring the NAT are managed by the routeOperator.

**Comment by palexster** [comment]
@alacuku can you please squash your commits?

**Pull request merged**